 0016-capi-upgrades/0016-capi-upgrades.md

```diff
@@ -0,0 +1,33 @@
+# RFC 016 - CLUSTER API UPGRADES
+
+## Reasoning
+
+As part of the Cluster API (CAPI) Hive Sprint, the `upgrades hive` decided to create a document to gather all actions need to be done in order to upgrade a workload cluster successfully as well as bring the functionality we have already built to trigger the upgrades.
+
+## Scope
+
+This document only cover the action to upgrade the workload cluster without solving problems of versioning of the different CAPI controllers and/or deciding which controller reconcile a cluster.
```

Comment:
I'd also make clear which upgrades we are focusing on.

A workload cluster can be ugpraded in different ways
- apps installed in the cluster are upgraded (out of scope here)
- cluster is reconciled by a newer CAPI controller
- machine image used by control plane or workers is updated with new k8s version or new OS version

#### Apps installed in the workload cluster
This upgrade is out of scope, and we have other non-CAPI components that already take care of that, like app-operator.

#### Reconcile Cluster with new CAPI controller release
We may want to upgrade the Cluster to a newer CAPI controller release, which may change the infrastructure created to support the cluster (network, load balancers, etc). We do this update just changing the CAPI label on the Cluster CR, and we don't need a new machine image to be created/referenced by the CR's.

#### Upgrade k8s version or OS version
Finally we may want to update the k8s version or the OS version running in the machines. In both cases, we need a new machine image using the updated components, and reference that image from the CR's.